lib/rdl/types/method.rb

```diff
@@ -52,18 +52,35 @@ def initialize(args, block, ret)
       super()
     end
 
+    def get_block()
+	return @block
+    end
+
     def le(other, h={})
       raise RuntimeError, "should not be called"
     end
 
     # TODO: Check blk
-    def pre_cond?(inst, *args, &blk)
+    def pre_cond?(blk, slf, inst, *args)
+	if @block then
+		blk = block_wrap(slf,inst,@block,&blk)
+	end
+	new_args = []
+	new_arg_typs = []
+	@args.each_with_index {|a,i| if a.is_a?(RDL::Type::MethodType) then
+					new_arg_typs << RDL::Type::NominalType.new('Proc')
+					new_args << block_wrap(slf,inst,a,&args[i])
+					else
```

Comment:
oh this spacing so ugly (just saying - as a random reader, no repository maintainer)